Edge impulse module

eoinjordan opened the pull request:
This pull request adds new documentation for integrating the Edge Impulse SDK as a Zephyr external module. The documentation provides an overview, setup instructions, usage guidance, and reference links to help users add and use the Edge Impulse SDK in Zephyr-based projects.

New documentation for Edge Impulse SDK integration:

* Added a new file `doc/develop/manifest/external/edge-impulse.rst` that explains how to add the Edge Impulse SDK as a Zephyr module, including manifest configuration, usage of west extension commands, and links to further tutorials and references.

https://builds.zephyrproject.io/zephyr/pr/101581/docs/develop/manifest/external/edge-impulse.html

kartben commented on doc/develop/manifest/external/edge-impulse.rst:
```diff
@@ -0,0 +1,72 @@
+The module can also provide west extension commands for working with Edge Impulse
+model artifacts.
+
+
```
```suggestion
```

kartben commented on doc/develop/manifest/external/edge-impulse.rst:
```diff
@@ -0,0 +1,72 @@
+Overview
+************
+
+Edge Impulse is a development platform for machine learning on edge devices. The Edge Impulse SDK for Zephyr
```
please set your editor to wrap lines at 100 columns (some are too long and some are really short - so it's also about consistency throughout the file - thanks!)

kartben commented on doc/develop/manifest/external/edge-impulse.rst:
```diff
@@ -0,0 +1,72 @@
+- ``west ei-deploy``: Downloads pre-built deployment artifacts
+- Both require ``-k`` (API key) and ``-p`` (project ID) flags
+
+**Note:** Commands must be run from the manifest repository directory and require ``west-commands`` registration in ``west.yml``.
```
not sure this note is needed, the command registration part is already shown in the snippet and it's fair to assume people would work from their workspace or would have set ``ZEPHYR_BASE`` to point to their current workspace.

eoinjordan commented on doc/develop/manifest/external/edge-impulse.rst:
```diff
@@ -0,0 +1,60 @@
+     projects:
+       - name: edge-impulse-sdk-zephyr
+         url: https://github.com/edgeimpulse/edge-impulse-sdk-zephyr
+         revision: v1.0.0
```
good point added

kartben reviewed (commented):
@eoinjordan please squash all commits into just one - and do one last reformatting pass to limit lines to 100 columns if you have a chance :) Thanks!

eoinjordan committed:
doc: add Edge Impulse SDK external module documentation

Add documentation for the Edge Impulse SDK Zephyr module to the external
modules section, including:

- Module overview describing the Edge Impulse ML development platform
- Integration instructions with west.yml manifest configuration
- West extension commands (ei-build, ei-deploy) for model deployment
- License information (BSD-3-Clause-Clear)
- References to official Edge Impulse documentation and tutorials

The Edge Impulse SDK module enables machine learning inference on Zephyr
devices by packaging the Edge Impulse SDK as a reusable Zephyr module
with west integration for streamlined ML model deployment workflows.

Signed-off-by: Eoin Jordan <eoin@edgeimpulse.com>
